### PR TITLE
Using a Better Method for the Log File

### PR DESCRIPTION
**Pull request by shayanfpg9**
**What kind of change does this PR introduce?**
I changed the [log.js](https://github.com/shayanfpg9/webpack/blob/main/hot/log.js) variables to ES6 and
I also used fewer lines of code to make it perform better
related to #16400

**Did you add tests for your changes?**
no

**Does this PR introduce a breaking change?**
Nope, this is the same code as the first but it is more modern and also uses fewer lines of code

**What needs to be documented once your changes are merged?**
nothing

### CODE REVIEW

File: hot/log.js [hunk 1]
```diff
@@ -1,13 +1,9 @@
-var logLevel = "info";
+let logLevel = "info";
```
Comment (alexander-akait) [inline hot/log.js:1]: We can't use `let` due compatbility, because we still suppport E5

File: hot/log.js [hunk 2]
```diff
@@ -47,11 +40,10 @@ module.exports.setLogLevel = function (level) {
 };
 
 module.exports.formatError = function (err) {
-	var message = err.message;
-	var stack = err.stack;
-	if (!stack) {
-		return message;
-	} else if (stack.indexOf(message) < 0) {
+	const {message , stack} = err;
```
Comment (alexander-akait) [inline hot/log.js:43]: Same for destructuring

File: hot/log.js [hunk 3]
```diff
@@ -47,11 +40,10 @@ module.exports.setLogLevel = function (level) {
...
+
+	if (!stack) return message;
+	if (stack.includes(message)) {
```
Comment (alexander-akait) [inline hot/log.js:46]: includes doesn't exists in old browsers

File: hot/log.js [hunk 4]
```diff
@@ -3,11 +3,7 @@ var logLevel = "info";
 function dummy() {}
 
 function shouldLog(level) {
-	var shouldLog =
-		(logLevel === "info" && level === "info") ||
-		(["info", "warning"].indexOf(logLevel) >= 0 && level === "warning") ||
-		(["info", "warning", "error"].indexOf(logLevel) >= 0 && level === "error");
-	return shouldLog;
+	return typeof level === "string" && ["info", "warning", "error"].includes(level);
```
Comment (alexander-akait) [inline hot/log.js:6]: includes can't be used, old browsers do not support it

Comment (shayanfpg9) [reply hot/log.js:6]: oh i forgot it 🤨🙄

**Pull request closed without merging**